src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1406,9 +1383,25 @@ public void visitMethodDef(JCMethodDecl tree) {
         }
 
         void analyzeSymbol(JCTree tree) {
-            if (isInLHS && !insideLambdaOrClassDef) return;
-            tree = TreeInfo.skipParens(tree);
             Symbol sym = TreeInfo.symbolFor(tree);
+            if (isInLHS && !insideLambdaOrClassDef) {
```

Comment:
Question: for stuff that is NOT in LHS -- shouldn't we check that preview features are enabled? E.g. we don't want to allow reads in prologue unless `--enable-preview` is present, right?